src/main/java/ch/naviqore/service/impl/TypeMapper.java

```diff
@@ -97,11 +97,45 @@ public static Leg map(ch.naviqore.raptor.Connection.Leg leg, LocalDate date, Gtf
         return switch (leg.type()) {
             case WALK_TRANSFER -> new TransferImpl(distance, duration, toLocalDateTime(leg.departureTime(), date),
                     toLocalDateTime(leg.arrivalTime(), date), sourceStop, targetStop);
-            // TODO: Refactor Raptor and extract interfaces. Put Trip id on leg, then stop time can be found.
-            case ROUTE -> new PublicTransitLegImpl(distance, duration, null, null, null);
+            case ROUTE -> createPublicTransitLeg(leg, date, schedule);
         };
     }
 
+    private static Leg createPublicTransitLeg(ch.naviqore.raptor.Connection.Leg leg, LocalDate date,
+                                              GtfsSchedule schedule) {
+        int duration = leg.arrivalTime() - leg.departureTime();
+        ch.naviqore.gtfs.schedule.model.Trip gtfsTrip = schedule.getTrips().get(leg.tripId());
+        Trip trip = map(gtfsTrip, date);
+
+        assert gtfsTrip.getStopTimes().size() == trip.getStopTimes()
```

Comment:
When or how could this be the case?

Reply:
I think this should never be the case... but since we compare datasets from two different sources (GTFS and Raptor) it could be that we somehow insert inconsistencies. Therefore I added the assert. In production asserts are not executed, so it is really just for us developer to ensure this consistency. Since I guess it could lead to an index out of bounds exception in the best case, in the worst case, the resulting leg is just incorrect without an error.